api/src/Server.ts

```diff
@@ -1,9 +1,11 @@
 import express from 'express';
 import { Request, Response } from 'express';
+import { getDbClientInstance } from './database/dbClient';
+import ProjectService from './projects/ProjectService';
 
 // Init express
 const app = express();
-
```

Comment:
This is a big improvement!
This pattern relies on default constructor args and a lazy singleton from the database client instance to simplify things. Though you can see that it would be pretty easy to just construct all of your objects here:
```
const dbClient = // Knex(....)
const projectDao = ProjectDao(dbClient)
const projectService = ProjectService(projectDao)
```

Next time we chat we can talk about dependency injection containers as an end goal of this type of pattern: https://stackoverflow.com/questions/50718586/what-is-a-di-container

Reply:
Let's talk about this today.